pkg/utils/utils.go

```diff
@@ -43,3 +44,8 @@ func (b *Broadcaster) Notify(instance interface{}) {
 func CloseCloserDisregardError(handler io.Closer) {
 	_ = handler.Close()
 }
+
+// MatchAddressFamily compares 2 addresses families and returns true if they're the same, else false
+func MatchAddressFamily(x net.IP, y net.IP) bool {
+	return x.To4() != nil && y.To4() != nil || x.To16() != nil && x.To4() == nil && y.To16() != nil && y.To4() == nil
```

Comment:
This is a neat 1-liner, but its really terse to read. It would probably be better broken out into individual if statements so it was more readable.